library/src/test/java/com/a99/rxplaces/RxAutocompleteStressTest.kt

```diff
@@ -0,0 +1,159 @@
+  }
+
+  @Test
+  fun manyInputs_differentPace_oneQuery() {
+    // given
+    val testSubscriber = TestSubscriber<Any>()
+    val testSubject = TestSubject.create<String>(testScheduler)
+    val rxAutocomplete = createRxAutoComplete(repository)
+
+    rxAutocomplete.observe(testSubject)
+        .subscribe(testSubscriber)
+
+    // when
+    testSubject.onNext("aven")
+    shiftTime(300, TimeUnit.MILLISECONDS)
+    verify(repository, never()).query(any(), any(), any())
+
+    testSubject.onNext("aveni")
+    shiftTime(500, TimeUnit.MILLISECONDS)
+    verify(repository, never()).query(any(), any(), any())
+
+    testSubject.onNext("avenida")
+    shiftTime(600, TimeUnit.MILLISECONDS)
+    verify(repository, never()).query(any(), any(), any())
+
+    testSubject.onNext("avenida bra")
+    shiftTime(600, TimeUnit.MILLISECONDS)
+    verify(repository).query(any(), any(), any())
+
+    // then
+    testSubscriber.assertNoErrors()
+    testSubscriber.assertValueCount(1)
+  }
+
+  @Test
+  fun manyInputs_differentPace_manyQueries() {
+    // given
+    val testSubscriber = TestSubscriber<Any>()
+    val testSubject = TestSubject.create<String>(testScheduler)
+    val rxAutocomplete = createRxAutoComplete(repository)
+
+    rxAutocomplete.observe(testSubject)
+        .subscribe(testSubscriber)
+
+    // when
+    testSubject.onNext("aven")
+    shiftTime(1, TimeUnit.SECONDS)
+    verify(repository, never()).query(any(), any(), any())
+
+    testSubject.onNext("aveni")
+    shiftTime(700, TimeUnit.MILLISECONDS)
+    verify(repository, never()).query(any(), any(), any())
+
+    testSubject.onNext("avenida")
+    shiftTime(500, TimeUnit.MILLISECONDS)
+    verify(repository).query(any(), any(), any())
+
+    testSubject.onNext("avenida bra")
+    shiftTime(2, TimeUnit.SECONDS)
+    verify(repository, times(2)).query(any(), any(), any())
+
+    // then
+    testSubscriber.assertNoErrors()
+    testSubscriber.assertValueCount(2)
+  }
+
+  @Test
+  fun randomInputs_inShortRandomInterval() {
+    val testSubscriber = TestSubscriber<Any>()
+    val randomInterval = { Random().nextInt(1000).toLong() }
+
+    val randomInputs = Observable.range(1, Int.MAX_VALUE)
+        .map { "Input $it" }
+        .delay { Observable.interval(randomInterval(), TimeUnit.MILLISECONDS) }
+
+    val repository = mock<PlacesAutocompleteRepository> {
+      on {
+        query(any(), any(), any())
+      } doReturn Single.fromCallable { listOf<Prediction>() }
+    }
+
+    val rxAutocomplete = createRxAutoComplete(repository)
+
+    rxAutocomplete.observe(randomInputs)
+        .subscribe(testSubscriber)
+
+    // when
+    shiftTime(10, TimeUnit.SECONDS)
+
+    // then
+    testSubscriber.assertNoErrors()
+    testSubscriber.assertValueCount(5)
+
+    verify(repository, times(5)).query(any(), any(), any())
+  }
+
+  private fun createRxAutoComplete(repository: PlacesAutocompleteRepository) = RxAutocomplete.create(testScheduler, repository, testSchedulerRule::logger)
```

Comment:
We can easily create an `Helper` inside the `test` package and move this guy to there